### ISSUE TITLE
Any node.js wizards want to lay out an initial project framework?

### ISSUE THREAD

**Issue by amoffat** [chaosbot/chaos#565]
I'm not experienced in node.js, so I was hoping to learn from some of you.  Any node.js guys want to create a new repo and put down some scaffolding?

**Comment by andrewda** [comment]
I can definitely work on that with someone

**Comment by PlasmaPower** [comment]
I'm currently working on some other stuff, but I'll likely have some free time next week or earlier. I'd love to help then.

**Comment by andrewda** [comment]
What do we want to call the repo? Are we going to be using any particular naming format?

**Comment by Redmega** [comment]
Count me in on the node team. This weekend I'll have some time to contribute

**Comment by mdcfe** [comment]
I'm currently busy, but I might be able to find some time to help.

**Comment by Swizz** [comment]
JS dev here too.
I am lacking motivation in these past day. But I can be ready to go when the hype will be a ton motivational.

**Comment by Redmega** [comment]
We should call it `chaos-node` if it's to be the central node module.

**Comment by Swizz** [comment]
@Chaosthebot/js-wizards team created :tada:

**Comment by rhengles** [comment]
> I'm currently busy, but I might be able to find some time to help.

^ This, also. I think it would help to have a overview of what needs to be implemented, since it would be hard for me to simply translate the Python source. Do we need a simple wrapper around [node-github](https://github.com/mikedeboer/node-github)?

**Comment by Redmega** [comment]
I think the first step is to replicate the core democracy functionality and then expand after that. I'd imagine the Node core would be ideal for any sort of web server needs, api endpoints, mongo/couch/nosql integration, etc.

I'd also love for us to use graphql. I've been using it at work and it's just plain awesome.

**Comment by mdcfe** [comment]
@Redmega I'd much rather keep the critical code in the core (updating bot from git) and isolate other non-critical components (web server, PR handling, issue/project commands) to other processes - [this project](https://github.com/orgs/Chaosthebot/projects/1) shows roughly how I thought functionality should be split.

**Comment by Redmega** [comment]
Let me stop using the word "Core" everywhere. Auto updating and restarting would be controlled by `chaos-core`. By "Core democracy functionality" I meant the PR voting, commands, etc. So in `chaos-core` we could specify that the `democracy_module` would point to `chaos-node/democracy` or something like that...

**Comment by mdcfe** [comment]
I've now created the Chaosthebot/Chaos-Core repo.
 
@Redmega It seems like the Node repo we're creating first is going to *be* the core, unless someone wants to write it in Python or another language instead. I don't necessarily think we should bundle together different parts of the bot in one repo per language.

**Comment by Redmega** [comment]
Maybe it might be better to separate via branch. `Chaos-Core/master-node` vs `Chaos-Core/master-python`?

**Comment by PlasmaPower** [comment]
I don't think per branch is a good idea. Branches usually separate versions of the same code.

**Comment by andrewda** [comment]
Yea if we are going to do different languages, it's got to be in different repos.

**Comment by mpnordland** [comment]
So if we write this in nodejs, how do we plan to allow other modules (in Rust, Go, Python, etc) to launch and communicate?
@md678685 That link you have is 404'ing  for me.

**Comment by mdcfe** [comment]
@mpnordland Child processes and IPC, but we'll need a standardised protocol for IPC and a format for knowing how to launch them. I've fixed the link.

**Comment by Redmega** [comment]
Maybe a modules.json which can be read to register modules map them to repos/endpoints/what-have-you.

**Comment by Redmega** [comment]
Maybe we can use something like `forever-monitor` to facilitate this.

**Comment by edpz** [comment]
Yeah  I think it's  a good idea

**Comment by edpz** [comment]
[

](url)
![2017_06_24_15_44_52](https://user-images.githubusercontent.com/29014568/27615964-b14e39a8-5bde-11e7-831d-7ca499febad2.png)

**Comment by edpz** [comment]
![bitcoin_16u8ij4zgzwy94qu](https://user-images.githubusercontent.com/29014568/27615994-f10f48a2-5bde-11e7-9bc0-021ff75c8505.png)
![29014568-1](https://user-images.githubusercontent.com/29014568/27616004-ffa47e0a-5bde-11e7-8405-c1ba8ae67143.png)

**Comment by andrewda** [comment]
@edpz Are those just bitcoin wallets? Please don't post things here that don't contribute to the discussion.

**Comment by PlasmaPower** [comment]
@andrewda Yeah I checked and they were BTC payment requests. I've removed the comments.

@edpz Further spam will result in a ban.

**Comment by PlasmaPower** [comment]
@amoffat Actually can we just ban @edpz? At the time I wrote my previous comment I hadn't read all my notifications, but they've been spamming all over the place.

**Comment by amoffat** [comment]
Looks like Github nuked them already.

**Comment by PlasmaPower** [comment]
Wow, that was fast! I reported them only an hour ago (when I wrote the comment).

**Comment by Redmega** [comment]
I reported them last night ;)

**Comment by Redmega** [comment]
Going to delete all the preceding comments about our spammer to keep the thread on topic
